engine/table/src/main/java/io/deephaven/engine/table/impl/asofjoin/RightIncrementalAsOfJoinStateManagerTypedBase.java

```diff
@@ -262,15 +266,19 @@ protected void buildTable(
             final ColumnSource<?>[] buildSources,
             final IntegerArraySource hashSlots,
             final TypedHasherUtil.BuildHandler buildHandler) {
+
+        // store this for access by the hashing methods
+        this.hashSlots = hashSlots;
```

Comment:
I think you should re-set this to `null` on leaving `buildTable`. Firstly, this way we don't keep the state manager's `slots` around until the first update. Secondly, I think it might help us identify bugs in maintenance of this field.